### DISCUSSION TITLE
How to link iOS native module to example?

### DISCUSSION THREAD

**Discussion by cbrzn** [callstack/react-native-builder-bob#437] [Q&A] [answered]
Hello guys I am trying to achieve a linking of my iOS project into the `example` folder but it's not clear to me how to achieve this. I read that you recommend that this should be done following this:
```
Linking the native code

By default, React Native CLI only links the modules installed under node_module of the app. To be able to link the android and ios folders from the project root, the path is specified in the example/react-native.config.js file.
```
But I am not sure how should I specify the path in this file? Does someone has an example?

If it helps, this is how I am doing the integration: https://github.com/polywrap/react-native/pull/2 - If I understand correctly once I add this path to the `react-native.config.js` I should just run `yarn ios` in the `example` folder and it should work? (If my bindings are correct of course but I first need to get tot his step)

**Comment by satya164** [comment, marked as answer]
All of this is already configured in the project generated by `create-react-native-library`. There's nothing additional to do.

> I read that you recommend that this should be done following this

Where did you read that we recommend that you should do this? The sentence you have quoted is an explanation of how it works in the generated project, not what you should do after generating the project.

**Reply by cbrzn** [reply]
thank you for your answer 🙏

**Reply by cbrzn** [reply]
I have another question, not sure if I should create another post or not; so if you think I should just let me know and we can continue the discussion there (I just dont want to spam your forum unless necessary). 

Let's say I want to create a native module that depends on a pod. This means that, in the `ios` folder of the root I would like to create a `Podfile` that has my pod as dependency. Is this a supported approach? Or even, I am thinking things in the correct way? Or should the consumer of my library add this dependency in the `Podfile` (This would mean that in the already existing `Podfile` of `examples/ios` I would add it)?

**Reply by atlj** [reply]
To add pod dependencies on your ios library, you can use the generated `your_project_name.podspec` file. 

You can simply:

```ruby
  s.dependency "React-Core"
  s.dependency "Your-Dependency" # You can add your pod dependency like this
```

The `Podfile` only affects your app. Namely the example app in your case.

**Reply by cbrzn** [reply]
thanks for your attention ser @atlj. Okay this makes sense (In advance sorry if my questions are ignorant, not proefficient with RN world). After I did this, I have noticed that my dependency is not being added into the `Pods` folder; it's worth mentioning that after doing this change I am doing `yarn` in the root folder and then in the `example` folder I am running `yarn ios` (I also deleted the `.lock` file & `Pods`), but the file in the `ios` folder (of the root) is not recognising my library. Am I missing any step? For reference, [this is how I am doing the change](https://github.com/cbrzn/react-native-native-ios/commit/ac4175a6e04b4eaa1c6fc0276b2b38acfab525b4#diff-b96453c450b2f4dfd48ea54b157bd4dee3d93b74d27824704eea8e9a53227476)

I also would like to mention that this library brings up a static framework (this is an [example of how I am using this in another pod]( https://github.com/polywrap/ethereum-wallet/blob/main/MetamaskProviderPlugin.podspec#L15-L16))

Again, thank you for your time and attention 🙏

**Reply by cbrzn** [reply]
All right I was able to fix it, I think the problem was the order on how I was doing things:

```
  s.dependency "PolywrapClient"

  # Use install_modules_dependencies helper to install the dependencies if React Native version >=0.71.0.
  # See https://github.com/facebook/react-native/blob/febf6b7f33fdb4904669f99d795eba4c0f95d7bf/scripts/cocoapods/new_architecture.rb#L79.
  if respond_to?(:install_modules_dependencies, true)
    install_modules_dependencies(s)
  else

  s.dependency "React-Core"
```
This works now ^_^

Once again, thank you for your time. Really appreciated

**Reply by atlj** [reply]
Awesome, glad it's working for you 🥳
